jadx-core/src/main/java/jadx/core/utils/files/ZipFile.java

```diff
@@ -0,0 +1,201 @@
+package jadx.core.utils.files;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.lang.reflect.UndeclaredThrowableException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
+public class ZipFile extends java.util.zip.ZipFile {
+
+	public ZipFile(File file) throws IOException {
+		this(file, OPEN_READ);
+	}
+
+	public ZipFile(File file, int mode) throws IOException {
+		this(file, mode, StandardCharsets.UTF_8);
+	}
+
+	public ZipFile(String name, Charset charset) throws IOException {
+		this(new File(name), OPEN_READ, charset);
+	}
+
+	public ZipFile(String name) throws IOException {
+		this(name, StandardCharsets.UTF_8);
+	}
+
+	public ZipFile(File file, int mode, Charset charset) throws IOException {
+		super(patchZipFile(file), mode, charset);
+	}
+
+	private static File patchZipFile(File file) throws IOException {
+		if (!file.getPath().toLowerCase().endsWith(".apk")) {
+			return file;
+		}
+
+		var raFile = new RandomAccessFile(file, "r");
```

Comment:
`raFile` is never closed. Better use a try-with-resource environment to auto-close it. 
The same also applies to `newRaFile`.